Index: src/java/com/eviware/soapui/impl/wsdl/mock/WsdlMockService.java
===================================================================
--- src/java/com/eviware/soapui/impl/wsdl/mock/WsdlMockService.java	(wersja 12844)
+++ src/java/com/eviware/soapui/impl/wsdl/mock/WsdlMockService.java	(wersja 12851)
@@ -103,7 +103,7 @@
 			config.setPath( "/" );
 
 		if( !getSettings().isSet( REQUIRE_SOAP_ACTION ) )
-			setRequireSoapAction( true );
+			setRequireSoapAction( false );
 
 		try
 		{
